packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -72,10 +72,6 @@ contract L2ToL2CrossDomainMessenger is IL2ToL2CrossDomainMessenger, ISemver {
     ///         message.
     uint240 internal msgNonce;
 
-    /// @notice Emitted whenever a message is sent to the other chain.
-    /// @param data Encoded data of the message that was sent.
-    event SentMessage(bytes data) anonymous;
```

Comment:
From an indexing perspective, having a non-anonymous event here is useful for querying logs. This approach might make this problematic? 

Any thoughts of the implications of this?

Reply:
Can you go into the specific usecase you are looking for a bit? What kind of indexing? You can express "give me all events emitted by this address" and "between these block numbers" and filter based on `log.data` for the sent message events

Reply:
Also this came up previously but didn't get a chance to follow up with a PR to the cross domain messenger.

Appropriately including the `msgNonce` in Sent/Relayed events so it's easy to correlate the two when inspecting on etherscan. With `RelayMessage` only emitting the message hash, it takes a serious amount of digging to figure out which initiating message it correlates with.

Reply:
Since the only event emitted by this contract is `SentMessage`, it would suffice to simply specify the contract address to get all events.

But typically `Topics[0]` would also be specified to include the event signature when filtering. If this contract was ever updated to include `SentMessage1` like the existing CDM, this would break filtering and make it non-trivial to determine the specific event.

With how it's structured now, you'd have to get all the events and inspect `log.data` to detect the 4 first bytes of `relayMessage()` fn selector instead of doing a simple `eth_getLogs` query

Reply:
> With RelayMessage only emitting the message hash, it takes a serious amount of digging to figure out which initiating message it correlates with.

`L2toL2 RelayedMessage(messageHash) event` should be paired with it's `CrossL2Inbox ExecutingMessage(_id, _messageHash) event`
that's the easiest way to find the event.

Reply:
I was hoping that you would be able to pass in an empty topic to filter for `LOG0` logs, but that does not appear to be the case given the following code: https://github.com/ethereum/go-ethereum/blob/67b81371008d147aa2ccebd3fa9655fe042b0f14/eth/filters/filter.go#L317

Happy to follow up with UX improvements in another PR

Reply:
yea that's fair, for interop events it should be much simpler to be able to inspect the `identifier` field unlike the existing L1/L2 CDM. Since `RelayedMessage` and `ExecutingMessage` events will follow each other in order, that'll be useful for indexing.

out of curiosity, is the gas saving to not include the sending nonce on the initiating significant?

Reply:
> out of curiosity, is the gas saving to not include the sending nonce on the initiating significant?

which nonce? `L2toL2` does send it's `nonce, version` and `CrossL2Inbox` has no nonce.

Reply:
the CDM msg nonce in the `RelayedMessage` and `SentMessage` events. The versioned hash isn't particularly useful when just looking at a relayed message event.

Reply:
@hamdiallam @skeletor-spaceman Lets move this convo to discord, I want to make sure there is an easy way to observe the cross chain messages and we haven't put a ton of thought into it to be honest